crates/crypto/bls/src/pubkey.rs

```diff
@@ -37,3 +39,19 @@ impl PubKey {
         self.inner.iter().as_slice()
     }
 }
+
+pub fn pubkey_from_str(key_str: &str) -> Result<PubKey, BLSError> {
+    let clean_str = key_str.strip_prefix("0x").unwrap_or(key_str);
+
+    let bytes = match decode(clean_str) {
+        Ok(b) => b,
+        Err(_) => return Err(BLSError::InvalidHexString),
+    };
+
+    if bytes.len() != 48 {
+        return Err(BLSError::InvalidByteLength);
+    }
+
+    let inner = FixedVector::from(bytes);
+    Ok(PubKey { inner })
+}
```

Comment:
You shouldn't implement the `impl FromStr for PubKey {}` trait, not do a custom function like this